Gems/SurfaceData/Code/Tests/SurfaceDataTest.cpp

```diff
@@ -602,16 +597,17 @@ TEST_F(SurfaceDataTestApp, SurfaceData_TestSurfacePointsFromRegion_ProviderModif
             &SurfaceData::SurfaceDataSystemRequestBus::Events::GetSurfacePointsFromRegion,
             regionBounds, stepSize, testTags, availablePointsPerPosition);
 
-        EXPECT_TRUE(ValidateRegionListSize(regionBounds, stepSize, availablePointsPerPosition));
-
         // We expect every entry in the output list to have two surface points (with heights 0 and 4),
         // and each point should have both the "test_surface1" and "test_surface2" tag.
         for (auto& pointList : availablePointsPerPosition)
         {
-            EXPECT_TRUE(pointList.size() == 2);
+            EXPECT_EQ(pointList.size(), 2);
+            float expectedZ = 4.0f;
             for (auto& point : pointList)
             {
+                EXPECT_EQ(point.m_position.GetZ(), expectedZ);
                 EXPECT_TRUE(point.m_masks.size() == 2);
```

Comment:
nit: you updated the above `EXPECT_TRUE(pointList.size() == 2);` to be EXPECT_EQ, should probably do the same with this.